src/pip/_internal/locations/_sysconfig.py

```diff
@@ -24,7 +24,7 @@
 
 _AVAILABLE_SCHEMES = set(sysconfig.get_scheme_names())
 
-_HAS_PREFERRED_SCHEME_API = sys.version_info >= (3, 10)
+_HAS_PREFERRED_SCHEME_API = 'get_preferred_scheme' in dir(sysconfig)
```

Comment:
```suggestion
_HAS_PREFERRED_SCHEME_API = bool(getattr(sysconfig, "get_preferred_scheme", None))
```

Maybe?

Reply:
Better still, `hasattr(sysconfig, "get_preferred_scheme")`, surely?

Reply:
Going in the getattr direction, I decided to save the value directly instead of its boolean value to make it easier to use.
Let me know if that's fine for you.

Reply:
@pfmoore  hasattr was definitly the proper way to do, I do not know what I originally had in mind.
@uranusjr  I've updated the PR to use getattr, let me know if hasattr is a preferred option.